### PR TITLE
Turtles

### PR DESCRIPTION
**Pull request by Ben10Omintrix**
## About The Pull Request
adds turtles to the game! but these aren't your typical turtles. 
turtlestates

these are flora-turtles, with giant trees growing on their shells. These trees can emit fields which affects nearby hydroponic plants. Initially, the trees start out as young buds, from there the tree can evolve into different types depending on what you feed the turtle.

Feeding them pesticides causes the tree to blossom to be purple. this tree's fields will help kill some pests and weeds in nearby plants.

Feeding them nutrients gives you the green tree, the fields will heal nearby plants

Feeding them mutators like uranium or left 4 zed gives you the yellow tree. the fields increase instability of plants

The turtle will emit these fields every once in a while, ONLY when its feeling happy. therefore you'll have to pet it, clean it and feed it every once in a while to keep it satisfied. You can view the turtle's happiness by shift clicking it.

https://github.com/user-attachments/assets/a47136a1-06a1-419e-acc2-2f6f4468e296

The turtle only eats seeds. after eating a seed, itll process it and spit out its corresponding fruit! (for example, feeding it an apple seed gives you an apple). itll also sometimes playfully headbutt your legs and it loves going around smelling the scent from nearby plants

you can get these turtles by fishing the hydroponics tray or by ordering them through cargo. 

## Why It's Good For The Game
adds a new fun way for botanists to take care of their plants. While these turtles alone arent enough to fully replace plant dedicated nutrients, they add small extra support. 

## Changelog
:cl:
add: adds flora-turtles. obtainable through cargo or by fishing from the hydroponics tray
/:cl:

### CODE REVIEW

File: sound/mobs/non-humanoids/turtle/leaves_shaking.ogg
Comment (Sadboysuss) [inline sound/mobs/non-humanoids/turtle/leaves_shaking.ogg:1]: needs attribution

**Review by Wallemations — APPROVED** [review]
Turt :)

**Comment by MrMelbert** [comment]
Turtwig was the best Gen 4 starter I agree

File: code/modules/mob/living/basic/turtle/turtle.dm [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+/mob/living/basic/turtle/proc/set_plant_growth(evolution_path, amount)
+	path_growth_progress[evolution_path] += amount
+	if(path_growth_progress[evolution_path] >= REQUIRED_TREE_GROWTH)
+		evolve_turtle()
+
+/mob/living/basic/turtle/examine(mob/user)
+	. = ..()
+
+	if(stat == DEAD)
+		. += span_notice("its tree seems to be all withered...")
```
Comment (MrMelbert) [inline code/modules/mob/living/basic/turtle/turtle.dm:105]:
```suggestion
		. += span_warning("Its tree seems to be all withered...")
```

File: code/modules/mob/living/basic/turtle/turtle.dm [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+
+	var/destined_path
+	var/current_max_growth = 0
+
+	for(var/evolution_path in path_growth_progress)
+		if(path_growth_progress[evolution_path] > current_max_growth)
+			destined_path = evolution_path
+			current_max_growth = path_growth_progress[evolution_path]
+
+	var/text_to_display = "its tree seems to be exuding "
```
Comment (MrMelbert) [inline code/modules/mob/living/basic/turtle/turtle.dm:116]:
```suggestion
	var/text_to_display = "Its tree seems to be exuding "
```

File: code/modules/mob/living/basic/turtle/turtle.dm [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+	. += living_tree
+
+/mob/living/basic/turtle/toggle_resting()
+	. = ..()
+	if(stat == DEAD)
+		return
+	icon_state = resting ? "[base_icon_state]_rest" : base_icon_state
+	regenerate_icons()
+
+/mob/living/basic/turtle/attackby(obj/item/used_item, mob/living/user, params)
```
Comment (MrMelbert) [inline code/modules/mob/living/basic/turtle/turtle.dm:175]: Should be item interaction probably?

File: code/modules/mob/living/basic/turtle/turtle.dm [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+	if(isnull(used_item.reagents))
+		balloon_alert(user, "empty!")
+		return
+
+	if(stat == DEAD)
+		balloon_alert(user, "its dead!")
+		return
+
+	var/should_transfer = FALSE
+	for(var/reagent as anything in path_requirements)
```
Comment (MrMelbert) [inline code/modules/mob/living/basic/turtle/turtle.dm:192]:
```suggestion
	for(var/reagent in path_requirements)
```

**Comment by DaydreamIQ** [comment]
Lets fucking GO TORTERRA

**Comment by Blacklist897** [comment]
this fits poorly with the art style

**Comment by Xkallubar** [comment]
This is not at all coherent with the art style.

**Comment by deathrobotpunch** [comment]
turt needs new sprites!

File: sound/mobs/non-humanoids/turtle/leaves_shaking.ogg
Comment (Ben10Omintrix) [reply sound/mobs/non-humanoids/turtle/leaves_shaking.ogg:1]: i ended up not using this sound at all anyway, just forgot to delete it

**Comment by TheVekter** [comment]
Goated

A big one you can ride around on would be cool as well

**Comment by chemistrymain3** [comment]
> this fits poorly with the art style

who the fuck cares it's torterra dude

**Comment by carlarctg** [comment]
would be cool if you could get non tree versions. maybe by feeding it plant b gone?

**Review by Wallemations — CHANGES_REQUESTED** [review]
> this fits poorly with the art style

Having a bit longer to think yea I think you're right
I was thinking about what could cause this and I think it has to do with the contrast.
It feels like the sprites were contrast-boosted, which makes them look odd.

Also, an important request I've got would be to update the turtle's outline so it's not just one shade of green all over.

File: code/modules/fishing/sources/source_types.dm [hunk 5]
```diff
@@ -481,6 +481,7 @@
 		/obj/item/seeds/random = 1,
 		/mob/living/basic/frog = 1,
 		/mob/living/basic/axolotl = 1,
+		/mob/living/basic/turtle = 2,
```
Comment (Ghommie) [inline code/modules/fishing/sources/source_types.dm:484]: Considering that's roughly a one in 3.8% chance of catching a turtle, I wonder if using a good/great quality bait should increase the chances of catching mobs a little? It should be as simple as adding some if checks to the `get_modified_table` proc.

**Comment by OneAsianTortoise** [comment]
dance-turtle

**Comment by KingkumaArt** [comment]
SPEED MERGE THIS NOW

**Review by Wallemations — APPROVED** [review]
Not perfect but better

File: code/modules/mob/living/basic/turtle/turtle.dm [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+
+/mob/living/basic/turtle
+	name = "turtle"
+	desc = "Dog."
+	icon_state = "turtle"
+	icon_living = "turtle"
+	icon_dead = "turtle_dead"
+	base_icon_state = "turtle"
+	icon = 'icons/mob/simple/pets.dmi'
+	butcher_results = list(/obj/item/food/meat/slab = 3, /obj/item/food/pickle = 1, /obj/item/stack/sheet/mineral/wood = 10)
```
Comment (Ghommie) [inline code/modules/mob/living/basic/turtle/turtle.dm:16]: Why does it leave a pickle behind when butchered? 😅

File: code/modules/mob/living/basic/turtle/turtle.dm [hunk 7]
```diff
@@ -0,0 +1,240 @@
...
+	var/last_direction = WEST
+
+/mob/living/basic/turtle/Initialize(mapload)
+	. = ..()
+
+	desc = pick(
+		"Likely Dog...",
+		"Praise the Dog!",
+		"Dog ahead.",
+		"Could this be a Dog?",
```
Comment (Ghommie) [inline code/modules/mob/living/basic/turtle/turtle.dm:60]: These are nice, however the full description should be a smidge more descriptive, hinting that trees can grow on these if possible.

**Review by Ghommie — APPROVED** [review]

**Comment by Ghommie** [comment]
Also sorry if I didn't review this earlier, it's been a bit busy for me this month.

File: code/modules/mob/living/basic/turtle/turtle.dm (hunk 6 above)
Comment (Ben10Omintrix) [reply code/modules/mob/living/basic/turtle/turtle.dm:16]: it was a silly reference to how they drop pickles on death in elden ring. but if its too out there i can remove it

File: code/modules/mob/living/basic/turtle/turtle.dm (hunk 7 above)
Comment (Ben10Omintrix) [reply code/modules/mob/living/basic/turtle/turtle.dm:60]: theres alot more details in /examine() .  if u feel anything's missing from there i can add it

File: code/modules/mob/living/basic/turtle/turtle.dm (hunk 6 above)
Comment (Ghommie) [reply code/modules/mob/living/basic/turtle/turtle.dm:16]: no no, it's good, I'm just don't play Elden Ring.

File: code/modules/mob/living/basic/turtle/turtle.dm (hunk 7 above)
Comment (Ghommie) [reply code/modules/mob/living/basic/turtle/turtle.dm:60]: Oh, I missed that, good then.

**Pull request merged**